src/Raven.Server/Smuggler/Documents/DatabaseSmuggler.cs

```diff
@@ -351,6 +338,73 @@ private SmugglerProgressBase.Counts ProcessIndexes(SmugglerResult result)
             return result.Indexes;
         }
 
+        private void WriteIndex(SmugglerResult result, IndexDefinition indexDefinition, IIndexActions actions, bool retryOnError)
+        {
+            try
+            {
+                if (_options.RemoveAnalyzers)
+                {
+                    foreach (var indexDefinitionField in indexDefinition.Fields)
+                        indexDefinitionField.Value.Analyzer = null;
+                }
+
+                actions.WriteIndex(indexDefinition);
+            }
+            catch (Exception e)
+            {
+                var exceptionMessage = e.Message;
+                if (exceptionMessage.Contains("No overload for method 'LoadDocument' takes 1 arguments"))
+                {
+                    exceptionMessage = $"{e.Message}" + Environment.NewLine +
+                            "LoadDocument requires a generic argument for its usage (the type of the loaded document)" + Environment.NewLine +
+                            "For example: " + Environment.NewLine +
+                                "\tfrom doc in doc.Orders" + Environment.NewLine +
+                                "\tlet company = LoadDocument(doc.Company, \"Companies\")" + Environment.NewLine +
+                                "\tselect new {" + Environment.NewLine +
+                                    "\t\tCompanyName: comapny.Name" + Environment.NewLine +
+                                "\t}" + Environment.NewLine;
+                }
+                else if (retryOnError && 
+                         (exceptionMessage.Contains("The name 'AbstractIndexCreationTask' does not exist in the current context") ||
+                          exceptionMessage.Contains("The name 'SpatialIndex' does not exist in the current context") ||
+                          exceptionMessage.Contains("The type or namespace name 'Abstractions' does not exist in the namespace 'Raven' (are you missing an assembly reference?)")))
+                {
+                    var maps = new HashSet<string>();
```

Comment:
I do not think we should try to fix the index on-fly.

Reply:
we can save the index as faulty index and allow the user to manually edit it.

Reply:
I do not think we want to do that, this is not a recommended way of migrating indexes.

Reply:
Yes, better to error clearly